spark-job-rest/src/main/scala/context/JobContextFactory.scala

```diff
@@ -0,0 +1,52 @@
+package context
+
+import api.ContextLike
+import com.typesafe.config.Config
+import org.slf4j.LoggerFactory
+
+import scala.util.Try
+
+trait JobContextFactory {
+  type C <: ContextLike
+  def makeContext(config: Config, contextName: String): C
+}
+
+object JobContextFactory {
+  val logger = LoggerFactory.getLogger(getClass)
+  val defaultFactoryClassName = "context.SparkContextFactory"
+  val classNameConfigEntry = "context.job-context-factory"
+
+  /**
+   * Loads context factory with a class name in `context.factory`
+   * @param config config for context and context factory
+   * @param contextName context name
+   * @return Spark context
+   */
+  def makeContext(config: Config, contextName: String): ContextLike = {
+    val factory = getFactory(config)
+    logger.info(s"Creating context $contextName from factory $factory.")
+    factory.makeContext(config, contextName)
+  }
+
+  /**
+   * Loads context factory from specified class
+   * @param className context factory class
+   * @return
+   */
+  def getFactory(className: String = defaultFactoryClassName): JobContextFactory = {
+    logger.info(s"Loading context factory $className.")
+    Class.forName(className).newInstance().asInstanceOf[JobContextFactory]
+  }
+
+  /**
+   * Loads factory for specified configuration.
+   * If config doesn't contain `context.job-context-factory` than [[JobContextFactory.defaultFactoryClassName]] will be loaded.
+   * @param config config
+   * @return
+   */
+  def getFactory(config: Config): JobContextFactory =
+    getFactory(Try {
+      config.getString(classNameConfigEntry)
```

Comment:
Instead of writing a Try each time you get the a config, you can use the generic method I have implemented, it is named getValueFromConfig.